### PR TITLE
ARROW-7515: [C++] Rename nonexistent and non_existent to not_found

### PR DESCRIPTION
**Pull request by mrkn**

### CODE REVIEW

**Comment by mrkn** [comment]
@nealrichardson Could you tell me the way to fix the following warning?

https://github.com/apache/arrow/pull/6139/checks?check_run_id=395390711#step:5:988
```
* checking for code/documentation mismatches ... WARNING
Codoc mismatches from documentation object 'open_source':
open_source
  Code: function(path, filesystem = c("auto", "local"), format =
                 c("parquet", "arrow", "ipc"), partitioning = NULL,
                 allow_not_found = FALSE, recursive = TRUE, ...)
  Docs: function(path, filesystem = c("auto", "local"), format =
                 c("parquet", "arrow", "ipc"), partitioning = NULL,
                 allow_non_existent = FALSE, recursive = TRUE, ...)
  Argument names in code not in docs:
    allow_not_found
  Argument names in docs not in code:
    allow_non_existent
  Mismatches in argument names:
    Position: 5 Code: allow_not_found Docs: allow_non_existent
```

**Comment by nealrichardson** [comment]
@mrkn edit r/man/open_source.Rd to match [the inline doc source you edited](https://github.com/apache/arrow/pull/6139/files#diff-2186b99dd95ca2ebd24d27b3eb44b319L214). Better would be to run `make doc` from the `r/` directory to regenerate the docs, though that requires that you have R set up and dependencies installed.

**Comment by wesm** [comment]
This needs to be rebased. @pitrou what do you think about this? I think it's okay

**Comment by mrkn** [comment]
@kou I added a commit for renaming things in c_glib.

**Comment by pitrou** [comment]
@mrkn Looks like the Ruby tests fails on Github Actions. You may be able to reproduce locally using:
```
docker-compose run ubuntu-ruby
```

**Review by kou — APPROVED** [review]
+1

I've pushed some more nonexistent to not-found changes.
And I've fixed GLib test failures.

**Comment by mrkn** [comment]
@kou thank you!

**Comment by kou** [comment]
I found more "nonexistent".
I've renamed them to "not found".
Python jobs will be green.

**Comment by mrkn** [comment]
@kou Thank you, again!

@pitrou Could you review this again?

File: cpp/src/plasma/test/client_tests.cc [hunk 1]
```diff
@@ -503,7 +503,7 @@ TEST_F(TestPlasmaStore, RefreshLRUTest) {
 TEST_F(TestPlasmaStore, DeleteTest) {
   ObjectID object_id = random_object_id();
 
-  // Test for deleting non-existence object.
+  // Test for deleting nonexistent object.
```
Comment (pitrou) [inline cpp/src/plasma/test/client_tests.cc]: Why these changes?

Comment (mrkn) [reply cpp/src/plasma/test/client_tests.cc]: The word "nonexistence" is a noun, so I guess "nonexistence object" means an object that represents the state of nonexistence.
On the other hand, the word "nonexistent" is an adjective, so I guess "nonexistent object" means an object that is absent.

Comment (pitrou) [reply cpp/src/plasma/test/client_tests.cc]: Ah, it didn't know it was actually a word. Thank you :-)
